### PR TITLE
Validation for max allowed number of chars in custom groups name

### PR DESCRIPTION
**Pull request by pako81**
## Description

Add validation for maximum allowed number of characters in custom groups names.

## Related Issue

- https://github.com/owncloud/enterprise/issues/3730

## Motivation and Context

Since we have already a check in place about the minimum number of chars a custom group name should be (2), it would make sense to also add a check for the maximum length, catch the exception and display a proper warning to the end-user. Currently, no warning is given back to the end-user and an exception is logged in the owncloud.log file.

## How Has This Been Tested?

Manually, by trying to create a new / rename an existing custom group having name longer than 64 chars and check that the above-mentioned exception is correctly caught and a proper warning is displayed to the end-user.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by dpakach** [comment]
@pako81 @phil-davis I have added some acceptance tests for the validation of custom group names length.

**Comment by pako81** [comment]
@dpakach thanks!

**Comment by pako81** [comment]
@dpakach codecov is failing

**Comment by phil-davis** [comment]
@pako81 Deepak has made acceptance tests, which are not part of code coverage. I guess that code coverage was failing the same before the acceptance tests were added.

@micbar can someone decide if code coverage needs improving? And if so, allocate an appropriate resource (developer?)

**Comment by phil-davis** [comment]
I rebased just now to get the PR up-to-date and fresh CI

**Comment by micbar** [comment]
@phil-davis Yes, indeed, we need more coverage.

Currently no dev available. Needs planning.

**Comment by phil-davis** [comment]
I added unit test cases for long group name validation. It turned out to be easy, and codecov is happy.
@micbar it just needs someone to review/approve/merge this PR.

**Comment by micbar** [comment]
@PVince81 We need some JS know how to review this. Could you? 😄

File: lib/Dav/GroupsCollection.php [hunk 1]
```diff
@@ -141,6 +141,11 @@ public function createExtendedCollection($name, Mkcol $mkCol) {
 			throw new ValidationException('The group name should be at least 2 characters long.');
 		}
 
+		/** Group name must be max 64 characters long */
+		if (\mb_strlen($name, 'UTF-8') > 64) {
+			throw new ValidationException('The group name should be maximum 64 characters long.');
+		}
```
Comment (jnweiger) [inline lib/Dav/GroupsCollection.php]: createGroup() is also called by public function createDirectory() -- should we validate the name there too?

**Pull request merged**